velox/functions/sparksql/tests/SplitTest.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/type/Type.h"
+
+using namespace facebook::velox::test;
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SplitTest : public SparkFunctionBaseTest {
+ protected:
+  void testSplit(
+      const std::vector<std::vector<std::string>>& expected,
+      const std::vector<std::string>& input,
+      std::string delim,
+      size_t numRows,
+      std::optional<int32_t> limit) {
```

Comment:
nit: adjust the order as (input, delim, limit, numRows, expected) for readability.